daemon/daemon_main.go

```diff
@@ -1566,3 +1584,29 @@ func initKubeProxyReplacementOptions() {
 		option.Config.EnableExternalIPs = false
 	}
 }
+
+func (d *Daemon) launchHubble() {
+	logger := logging.DefaultLogger.WithField(logfields.LogSubsys, "hubble")
+	addresses := option.Config.HubbleListenAddresses
+	if len(addresses) == 0 {
+		logger.Info("Hubble server is disabled")
+		return
+	}
+	epDNSGetter := hubble.NewLocalEndpointDNSGetter(d.endpointManager)
+	identityGetter := hubble.NewLocalIdentityGetter(d.identityAllocator)
+	ipGetter := hubble.NewLocalIPGetter(ipcache.IPIdentityCache)
+	serviceGetter := hubble.NewLocalServiceGetter(d.svc)
+	payloadParser, _ := parser.New(epDNSGetter, identityGetter, epDNSGetter, ipGetter, serviceGetter)
+	s := hubbleServer.NewLocalServer(payloadParser, option.Config.HubbleFlowBufferSize, logger)
+	go s.Start()
+	d.monitorAgent.GetMonitor().RegisterNewListener(context.TODO(), hubble.NewHubbleListener(s))
+	logger.WithField("addresses", addresses).Info("Starting Hubble server")
+	hubbleServe.Serve(logger, addresses, s)
```

Comment:
I'm concern about this. Cilium will be running on host network namespace. Having this will mean that hubble will expose all traffic metadata to anyone plus it will be running in under the same process of the control plane.

Reply:
The big question is how we will do multi-node. We may require TLS for IP based API usage to enable multi-node.

Reply:
should we limit this to unix domain socket and/or local interface while we figure this out?

Reply:
I would prefer to default to a unix domain socket for now.

Reply:
Also for this line: Please log the error returned by `hubbleServe.Serve`

Reply:
done!

@gandro by the way logging the error returned by `hubbleServe.Serve` was immediately useful :) the Serve() function was failing with unix domain socket because the hubble group doesn't exist. opened a pr: https://github.com/cilium/hubble/pull/123